src/lib/extract/extractOpacity.ts

```diff
@@ -1,6 +1,9 @@
 import type { NumberProp } from './types';
 
 export default function extractOpacity(opacity: NumberProp | void) {
+  if (typeof opacity === 'string' && opacity.trim().endsWith('%')) {
+    return +opacity.slice(0, -1) / 100;
```

Comment:
What about some weird % values as `500%`? It probably doesn't make sense but if we handle it with normal values then maybe we could to the same with %? If not, then we can just merge it as it is.

Reply:
Fixed in https://github.com/software-mansion/react-native-svg/pull/2325/commits/f63dcd7f2c192c9eb6b45dc403732ff8c78a5f01